lib/Target/CBackend/CBackend.cpp

```diff
@@ -4830,11 +4827,7 @@ void CWriter::visitCallInst(CallInst &I) {
 
   if (NeedsCast) {
     // Ok, just cast the pointer type.
-    Out << "((";
-    printTypeName(
-        Out, I.getCalledOperand()->getType()->getNonOpaquePointerElementType(),
-        false, std::make_pair(PAL, I.getCallingConv()));
-    Out << "*)(void*)";
+    Out << "((" << getFunctionName(&I) << "*)(void*)";
```

Comment:
Shouldn't this be a FunctionType here?

Reply:
No: this used to call `printTypeName`, which would call `getFunctionName` for a `FunctionType`, hence why I'm calling `getFunctionName` directly now using the `CallInst` (`printTypeName` can't print a `FunctionType` anymore since we lose the type information for anything hidden behind a pointer (struct returns, ByVal parameters, etc.)).

Reply:
Ah, I see. Yeah, it was just confusing that it sounded like it was getting the name of the function, rather than the name of the type of the function.